building h2o failed on Alpine with h2o v1.7.0-beta2

Hi,

Tried to build h2o with
```
cmake -DWITH_BUNDLED_SSL=off -DWITH_MRUBY=ON 
```

I'm getting error
```
-- The C compiler identification is GNU 5.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.28") 
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Found OpenSSL: /usr/lib/libssl.so;/usr/lib/libcrypto.so (found version "1.0.2e") 
-- Found ZLIB: /lib/libz.so (found version "1.2.8") 
-- checking for module 'libuv>=1.0.0'
--   found libuv, version 1.7.5
-- checking for module 'libwslay'
--   found libwslay, version 1.0.0
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/h2o
Scanning dependencies of target mruby
(in /tmp/h2o/deps/mruby)
make[2]: *** [CMakeFiles/mruby] Error 1
make[1]: *** [CMakeFiles/mruby.dir/all] Error 2
make: *** [all] Error 2
rake aborted!
cannot load such file -- rake/clean
Rakefile:15:in `load'
CMakeFiles/mruby.dir/build.make:57: recipe for target 'CMakeFiles/mruby' failed
CMakeFiles/Makefile2:408: recipe for target 'CMakeFiles/mruby.dir/all' failed
Makefile:127: recipe for target 'all' failed
```

but it is all fine with v1.7.0-beta1. am i missing something here?

Thanks

--- a/deps/mruby-file-stat/README.md
+++ b/deps/mruby-file-stat/README.md
@@ -1,7 +1,6 @@
 # mruby-file-stat
 
 [![Build Status](https://travis-ci.org/ksss/mruby-file-stat.svg?branch=master)](https://travis-ci.org/ksss/mruby-file-stat)
-
 [![Build status](https://ci.appveyor.com/api/projects/status/04am84uda2cj49n3/branch/master?svg=true)](https://ci.appveyor.com/project/ksss/mruby-file-stat/branch/master)
 
 **File::Stat** class in mruby
@@ -52,7 +51,8 @@ stat.setgid?
 stat.sticky?
 stat.ftype #=> socket, link, file, blockSpecial, directory, characterSpecial, fifo or unknown
 
-# support methods
+# Obsolete methods
+# Please use https://github.com/ksss/mruby-process-ext if you want to use these
 Process.uid
 Process.gid
 Process.euid
@@ -69,9 +69,16 @@ Write in /mruby/build_config.rb
 
 ```ruby
 MRuby::Build.new do |conf|
+  # by mgem
+  conf.gem :mgem => 'mruby-file-stat'
+  # by github
   conf.gem :github => 'ksss/mruby-file-stat', :branch => 'master'
 end
 ```
+
+## Homepage
+
+https://github.com/ksss/mruby-file-stat
 
 ## License
 

--- a/deps/mruby-file-stat/mrbgem.rake
+++ b/deps/mruby-file-stat/mrbgem.rake
@@ -1,5 +1,4 @@
 require 'mkmf'
-require 'rake/clean'
 
 # patch for old mkmf bug
 if RUBY_VERSION <= "1.9.3"
@@ -11,11 +10,8 @@ if RUBY_VERSION <= "1.9.3"
 end
 # patch end
 
-file_stat_dir = File.dirname(__FILE__)
-extconf = "#{file_stat_dir}/src/extconf.h"
-
-file extconf => ["#{file_stat_dir}/src/file-stat.c"] do |t|
-  File.unlink(t.name) if File.exist?(t.name)
+build_extconf = lambda do |fn|
+  return if File.exist?(fn)
 
   # TODO
   # if open this block
@@ -30,15 +26,15 @@ file extconf => ["#{file_stat_dir}/src/file-stat.c"] do |t|
     have_func "getgroups", "unistd.h"
   end
 
-  create_header t.name
+  create_header fn
 end
-
-CLOBBER << extconf
 
 MRuby::Gem::Specification.new('mruby-file-stat') do |spec|
   spec.license = 'MIT'
   spec.author  = 'ksss <co000ri@gmail.com>'
   spec.add_dependency('mruby-time')
 
-  Rake::Task[extconf].invoke
+  FileUtils.mkdir_p build_dir
+  build_extconf["#{build_dir}/extconf.h"]
+  cc.include_paths << build_dir
 end

--- a/deps/mruby-file-stat/src/file-stat.c
+++ b/deps/mruby-file-stat/src/file-stat.c
@@ -882,10 +882,22 @@ mrb_mruby_file_stat_gem_init(mrb_state* mrb)
   mrb_define_method(mrb, stat, "owned?", stat_owned_p, MRB_ARGS_NONE());
   mrb_define_method(mrb, stat, "owned_real?", stat_owned_real_p, MRB_ARGS_NONE());
 
-  mrb_define_class_method(mrb, process, "uid", process_getuid, MRB_ARGS_NONE());
-  mrb_define_class_method(mrb, process, "gid", process_getgid, MRB_ARGS_NONE());
-  mrb_define_class_method(mrb, process, "euid", process_geteuid, MRB_ARGS_NONE());
-  mrb_define_class_method(mrb, process, "egid", process_getegid, MRB_ARGS_NONE());
+  /*
+  Obsolete methods
+  Avoid install when already defined by other lib
+  */
+  if (!mrb_obj_respond_to(mrb, ((struct RObject *)process)->c, mrb_intern_lit(mrb, "uid"))) {
+    mrb_define_class_method(mrb, process, "uid", process_getuid, MRB_ARGS_NONE());
+  }
+  if (!mrb_obj_respond_to(mrb, ((struct RObject *)process)->c, mrb_intern_lit(mrb, "gid"))) {
+    mrb_define_class_method(mrb, process, "gid", process_getgid, MRB_ARGS_NONE());
+  }
+  if (!mrb_obj_respond_to(mrb, ((struct RObject *)process)->c, mrb_intern_lit(mrb, "euid"))) {
+    mrb_define_class_method(mrb, process, "euid", process_geteuid, MRB_ARGS_NONE());
+  }
+  if (!mrb_obj_respond_to(mrb, ((struct RObject *)process)->c, mrb_intern_lit(mrb, "egid"))) {
+    mrb_define_class_method(mrb, process, "egid", process_getegid, MRB_ARGS_NONE());
+  }
 }
 
 void

